riscv/csrs.h

```diff
@@ -475,25 +475,14 @@ class envcfg_csr_t: public masked_csr_t {
   virtual bool unlogged_write(const reg_t val) noexcept override;
 };
 
-// henvcfg.pbmte is read_only 0 when menvcfg.pbmte = 0
-// henvcfg.stce is read_only 0 when menvcfg.stce = 0
-// henvcfg.hade is read_only 0 when menvcfg.hade = 0
-// henvcfg.dte is read_only 0 when menvcfg.dte = 0
 class henvcfg_csr_t final: public envcfg_csr_t {
  public:
-  henvcfg_csr_t(processor_t* const proc, const reg_t addr, const reg_t mask, const reg_t init, csr_t_p menvcfg);
-
-  reg_t read() const noexcept override {
-    return (menvcfg->read() | ~(MENVCFG_PBMTE | MENVCFG_STCE | MENVCFG_ADUE | MENVCFG_DTE)) & masked_csr_t::read();
-  }
-
```

Comment:
Its unclear why these changes/restructuring was needed.

Reply:
Since that restructuring is unrelated to this PR, I suggest that this PR not do this. The refactoring/restructuring of code should be in a separate PR.